ls1intum/Artemis
# `Exam mode`: Improve behavior of exam cards for non participating students and tutors

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


Improve the interaction in edge cases of the exam card. These are students that did not participate and tutors that click on the exam card.
Fixes #6054.

### Description

The text on the exam card is now more streamlined such that it does not show confusing text to students that did not participate. I.E. "Click here to see the results of the exam".
Additionally the exam card no longer has a hover effect when it is not clickable, i.e. when the student cannot start another test exam.

When a tutor or higher clicks on the exam card there is no longer a server error shown. Instead the user has the option to go to the exam overview.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Exam

1. Log in to Artemis
2. Create the exam correctly
3. Try to click on the exam card as instructor and check the reroute to the overview
4. Check the text on the exam card as not participating student in different stages (before exam start but already visible, during exam conduction, after deadline of exam, after results are published)
5. Check that the exam card of a test exam no longer has a hover effect, when you participated in the test exam or when you missed the due date of the test exam.
6. Check that a participating student can always click on the exam card and sees the correct text for each stage of the exam conduction (e.g. imminent, running, review open, ...)

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-participation.component.ts | 68% | 80% |
| course-exam-detail.component.ts | 84% | 92% |
| ExamAccessService | 84% | 94% |

### Screenshots



View of a non-participating student after the exam due date:
![Bildschirm­foto 2023-02-23 um 15 49 17](https://user-images.githubusercontent.com/38322605/220942929-6c08749e-c502-4700-886f-e79db4771337.png)

View of a tutor after clicking on the exam card:
![Bildschirm­foto 2023-02-23 um 15 49 08](https://user-images.githubusercontent.com/38322605/220942940-abb2581b-681b-41f0-9e1a-d58f8ebf6497.png)
